src/hotspot/cpu/riscv/interp_masm_riscv.cpp

```diff
@@ -191,13 +198,36 @@ void InterpreterMacroAssembler::get_dispatch() {
 }
 
 void InterpreterMacroAssembler::get_cache_index_at_bcp(Register index,
+                                                       Register tmp,
                                                        int bcp_offset,
                                                        size_t index_size) {
   assert(bcp_offset > 0, "bcp is still pointing to start of bytecode");
   if (index_size == sizeof(u2)) {
-    load_unsigned_short(index, Address(xbcp, bcp_offset));
+    if (AvoidUnalignedAccesses) {
+      assert_different_registers(index, tmp);
+      load_unsigned_byte(index, Address(xbcp, bcp_offset));
+      load_unsigned_byte(tmp, Address(xbcp, bcp_offset + 1));
+      slli(tmp, tmp, 8);
+      add(index, index, tmp);
+    } else {
+      load_unsigned_short(index, Address(xbcp, bcp_offset));
+    }
   } else if (index_size == sizeof(u4)) {
-    lwu(index, Address(xbcp, bcp_offset));
+    if (AvoidUnalignedAccesses) {
+      assert_different_registers(index, tmp);
+      load_unsigned_byte(index, Address(xbcp, bcp_offset));
+      load_unsigned_byte(tmp, Address(xbcp, bcp_offset + 1));
+      slli(tmp, tmp, 8);
+      add(index, index, tmp);
+      load_unsigned_byte(tmp, Address(xbcp, bcp_offset + 2));
+      slli(tmp, tmp, 16);
+      add(index, index, tmp);
+      load_unsigned_byte(tmp, Address(xbcp, bcp_offset + 3));
+      slli(tmp, tmp, 24);
+      add(index, index, tmp);
+    } else {
```

Comment:
Does it really make sense to do this here? Shouldn't there be an unaligned version of `lwu` in MacroAssembler?

Reply:
Updated the PR, load_word_misaligned looks slightly ugly due to **Address -> (Address + offset)** conversion